azure_functions_worker/bindings/meta.py

```diff
@@ -25,11 +29,38 @@ def load_binding_registry() -> None:
     global BINDING_REGISTRY
     BINDING_REGISTRY = func.get_binding_registry()
 
-
-def get_binding(bind_name: str) -> object:
+    # The SDKs only support python 3.9+
+    if sys.version_info.minor > 8:
+        # Check if cx has imported sdk bindings library
+        try:
+            clients = importlib.util.find_spec('azure.functions.extension.base')
+        except ModuleNotFoundError:
+            # This will throw a ModuleNotFoundError in env reload because
+            # azure.functions.extension isn't loaded in
+            clients = None
+
+        # This will be None if the library is not imported
+        # If it is not None, we want to set and use the registry
+        if clients is not None:
+            import azure.functions.extension.base as clients
+            global SDK_BINDING_REGISTRY
+            SDK_BINDING_REGISTRY = clients.get_binding_registry()
+
+
+def get_binding(bind_name: str, pytype: typing.Optional[type] = None) -> object:
     binding = None
+
     registry = BINDING_REGISTRY
-    if registry is not None:
+    client_registry = SDK_BINDING_REGISTRY
+    # checks first if registry exists (library is imported)
+    # then checks if pytype is a supported type (cx is using sdk type)
+    if (client_registry is not None
+            and client_registry.check_supported_type(pytype)):
+        global deferred_bindings_enabled
+        deferred_bindings_enabled = True
```

Comment:
is it possible we just expose a single function from extension_base like deferred_bindings_enable(pytype)? i see this is checked in multiple places and a single checker function can make it simpler. you can also cache if the result like http_v2

```
class Cache:
    _http_v2_enabled: Optional[bool] = None

def http_v2_enabled() -> bool:
    if Cache._http_v2_enabled is None:
        Cache._http_v2_enabled = ModuleTrackerMeta.module_imported()
    return Cache._http_v2_enabled
```

Reply:
is there a better place to do this? get_binding is called multiple times right, deferred_bindings_enabled shall be set once is enough?

Reply:
Because we don't support 3.7 for the extensions but still do in the worker, I don't think adding a helper method in the base extension would improve the check much. We'd have to do a check to see if the base extension has been installed or on the python version e.g:
```
if base_ext is not None
   and base_ext.check_deferred_bindings_enabled():
        ....
```

Good point on the flag. I changed it so that the flag is now set only once, looking through to see if there's another place we can set it.

Reply:
In order to set the flag, we have to check the type declaration of every binding. If at least one type is declared as an sdk type, then the flag needs to be True. I moved setting it out from get_binding() to validate_function_params(). validate_function_params() is only called once and already does checks on the type declaration. Now, flag will only be set once for function apps using deferred bindings.